base/rational.jl

```diff
@@ -14,8 +14,16 @@ struct Rational{T<:Integer} <: Real
         num2, den2 = (sign(den) < 0) ? divgcd(-num, -den) : divgcd(num, den)
         new(num2, den2)
     end
+
+    function Rational{T}(num::Integer, den::Integer) where T<:BitSigned
+        num == den == zero(T) && __throw_rational_argerror(T)
+        num < 0 && den < 0 && (num === typemin(T) || den === typemin(T)) && __throw_rational_ovferror(T)
+        num2, den2 = (sign(den) < 0) ? divgcd(-num, -den) : divgcd(num, den)
+        new(num2, den2)
+    end
 end
 @noinline __throw_rational_argerror(T) = throw(ArgumentError("invalid rational: zero($T)//zero($T)"))
+@noinline __throw_rational_ovferror(T) = throw(OverflowError("num[den] is typemin(T) and den[num] is negative"))
```

Comment:
I find this error message a bit confusing: perhaps two different errors for the two cases? Or maybe just throw an overflow error? Or vaguer wording about one of the numerator or denominator being too large?

Reply:
Thanks for the note. Revised it:
```julia
@noinline __throw_rational_ovferror(num::T, den::T) where {T} =
    (num < den) ? throw(OverflowError("typemin($T)//$den")) : throw(OverflowError("$num//typemin($T)"))
```